app/models/piece.rb

```diff
@@ -1,6 +1,43 @@
 class Piece < ApplicationRecord
   belongs_to :game
+
+  def move_to!(new_x, new_y, white_turn) 
```

Comment:
Instead of passing `white_turn`.

If you call `self.is_white?`, you'll be able to see if the current piece that is being moved is white. Doing that will eliminate needing to pass in `white_turn`